larray_editor/arraywidget.py

```diff
@@ -841,18 +844,35 @@ def _data_digits(self, data, maxdigits=6):
         return maxdigits
 
     def autofit_columns(self):
+        self.view_axes.autofit_columns()
+        for column in range(self.model_axes.columnCount()):
+            self.resizeAxesColumnToContents(column)
         self.view_xlabels.autofit_columns()
         for column in range(self.model_xlabels.columnCount()):
-            self.resizeColumnToContents(column)
+            self.resizeDataColumnToContents(column)
 
-    def resizeColumnToContents(self, column):
+    def resizeAxesColumnToContents(self, column):
+        # must be connected to view_axes.horizontalHeader().sectionHandleDoubleClicked signal
+        width = max(self.view_axes.horizontalHeader().sectionSize(column),
+                    self.view_ylabels.sizeHintForColumn(column))
+        # no need to call resizeSection on view_ylabels (see synchronization lines in init)
+        self.view_axes.horizontalHeader().resizeSection(column, width)
+
+    def resizeDataColumnToContents(self, column):
```

Comment:
rename to resize_xlabels_column_to_contents because this works first on the view_xlabels not view_data? 
* to be consistent with resizeAxes... which works on view_axes
* to respect python naming conventions (this is not an overridden method name)